### PR TITLE
Fix FormCheckbox erroring when an "integer-like" key is used

### PR DESCRIPTION
**Pull request by Exanite**
This fixes https://github.com/ariakit/ariakit/issues/2799.

### CODE REVIEW

File: packages/ariakit-core/src/form/form-store.ts [hunk 1]
```diff
@@ -57,9 +53,9 @@ export function get<T>(
     return defaultValue as T;
   }
   if (!rest.length) {
-    return values[key] ?? defaultValue;
+    return (values[key] ?? defaultValue) as T;
```
Comment (Exanite) [inline packages/ariakit-core/src/form/form-store.ts]: Not sure if using the type assertions here is a good idea, but it doesn't affect functionality and there's no way to verify that the returned value is actually T.

File: packages/ariakit-core/src/form/form-store.ts [hunk 2]
```diff
@@ -18,15 +18,10 @@ import {
...
-  AnyObject,
-  PickRequired,
-  SetState,
-  SetStateAction,
-} from "../utils/types.js";
+import type { PickRequired, SetState, SetStateAction } from "../utils/types.js";
 import type { DeepMap, DeepPartial, Names, StringLike } from "./types.js";
 
-type Values = AnyObject;
+type Values = Record<string, unknown>;
```
Comment (diegohaz) [inline packages/ariakit-core/src/form/form-store.ts]: I'm thinking now that this could represent a breaking change since this `Values` type is exposed as `FormStoreValues` below.

We can do that in a separate PR, but it's probably safer for this PR to just include the `Array.isArray` fix.

**Review by diegohaz — COMMENTED** [review]
Thanks for working on this, @Exanite!

Would you mind [creating a basic example](https://github.com/ariakit/ariakit/blob/main/contributing.md#writing-another-example) (without a readme.md file) that illustrates your use case along with some tests? This will prevent future regressions.

File: packages/ariakit-core/src/form/form-store.ts (hunk 2 above)
Comment (Exanite) [reply packages/ariakit-core/src/form/form-store.ts]: Good catch. I'll remove these changes from my PR.
